apps/namadillo/src/atoms/balance/functions.ts

```diff
@@ -0,0 +1,58 @@
+import { IbcToken, NativeToken } from "@anomaorg/namada-indexer-client";
+import { Asset, AssetList } from "@chain-registry/types";
+import BigNumber from "bignumber.js";
+import { namadaAsset } from "registry/namadaAsset";
+import { TokenBalance } from "./atoms";
+
+// TODO upgrade this function to be as smart as possible
+// please refer to `tryCoinToIbcAsset` and how we could combine both
+export const findAssetByToken = (
```

Comment:
It seems that this function takes an IBC trace such as `transfer/channel-0/uatom` and checks if there is any known asset with the same trace on any chain. But I think this only works by coincidence because there happens to be a `channel-0` on Osmosis with counterparty Cosmoshub, and for other tokens we won't find anything. Or in other words, Namada does have a `channel-0` (corresponding to Cosmoshub testnet) but this is a different thing from `channel-0` on Osmosis (corresponding to Cosmoshub mainnet).

But I think we can leave this as is for now, and I can integrate the two different methods once #1228 is merged.

Reply:
Thanks! Exactly this! 🚀